modules/services/tomat.nix

```diff
@@ -0,0 +1,71 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.tomat;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  meta.maintainers = with lib.maintainers; [ jolars ];
+
+  options.services.tomat = {
+    enable = lib.mkEnableOption "Tomat Pomodoro server";
+
+    package = lib.mkPackageOption pkgs "tomat" { };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule { freeformType = tomlFormat.type; };
+
+      default = { };
+
+      example = {
+        timer = {
+          work = 25;
+          break = 5;
+          auto_advance = false;
+        };
+
+        sound = {
+          enabled = true;
+        };
+
+        notification = {
+          enabled = true;
+          sound = "Funk";
+        };
+      };
+
+      description = ''
+        Tomat configuration.
+        See <https://github.com/jolars/tomat/blob/main/docs/configuration.md> for supported values.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    xdg.configFile."tomat/config.toml".source = tomlFormat.generate "tomat-config.toml" cfg.settings;
```

Comment:
Usually prefer guarding this behind a check if a user has configured anything. Someone might want the package install and service, but use an unmanaged config file.